src/hash.h

```diff
@@ -194,6 +194,12 @@ uint256 SerializeHash(const T& obj, int nType=SER_GETHASH, int nVersion=PROTOCOL
     return ss.GetHash();
 }
 
+/** Single-SHA256 a 32-byte input (represented as uint256).
+ *  hash_io is read and written over with the result.
+ */
+void SHA256Uint256(uint256& hash_io);
+uint256 SHA256Uint256(uint256&& hash_io);
```

Comment:
What is the point of having these functions instead of a single `uint256 SHA256(const uint256& input)` ?

Reply:
One benefit is that both versions operate completely in-place on their arguments (otherwise we're reading from a pointer, writing to the stack, then writing to a pointer). Another benefit is that it is slightly harder to misuse this because when you are operating on a l-value there is no return value, which otherwise you might forget to assign.

Reply:
That could be solved using NODISCARD instead. I don't think saving a super brief 32 bytes on the stack is even going to be be observable.